File: ObjectCDemo/DatingApp/DatingApp/View/GuestCircleView/GuestCircleView.m
//
//  GuestCircleView.m
//  DatingApp
//
//  Created by Steven Jimy   on 11/11/16.
//  Copyright © 2016 Steven Jimy . All rights reserved.
//

#import "GuestCircleView.h"

@implementation GuestCircleView

- (instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        CGFloat width = frame.size.height - 10;
        self.nameLabel = [[UILabel alloc]initWithFrame:CGRectMake(5, 5, width, width)];
        self.nameLabel.backgroundColor = [UIColor clearColor];
        self.nameLabel.layer.cornerRadius = width/2;
        self.nameLabel.layer.borderColor = [UIColor whiteColor].CGColor;
        self.nameLabel.layer.borderWidth = 1.0f;
        self.nameLabel.textColor = [UIColor whiteColor];
        self.nameLabel.textAlignment = NSTextAlignmentCenter;
        [self.nameLabel setFont:[UIFont systemFontOfSize:10.0f]];
        self.nameLabel.numberOfLines = 0;
        [self addSubview:self.nameLabel];
    }
    return self;
}

- (void)setGuestModel:(GuestModel *)guestModel{
    _guestModel = guestModel;
    self.nameLabel.text = [guestModel getName];
}
@end

File: SwiftDemos/DatingApp/DatingApp/View/GuestCircleView/GuestCircleView.swift
//
//  GuestCircleView.swift
//  DatingApp
//
//  Created by Steven Jimy   on 11/11/16.
//  Copyright © 2016 Steven Jimy . All rights reserved.
//

import UIKit

class GuestCircleView: UIView {

    var nameLabel : UILabel!
    public var userModel : UserModel!
    
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        let width : CGFloat = frame.size.height - 10
        
        nameLabel = UILabel.init(frame: CGRect(x: 5, y: 5, width: width, height: width))
        nameLabel.backgroundColor = UIColor.clear
        nameLabel.layer.cornerRadius = width/2
        nameLabel.layer.borderColor = UIColor.white.cgColor
        nameLabel.layer.borderWidth = 1
        nameLabel.font = UIFont.systemFont(ofSize: 10)
        nameLabel.textAlignment = NSTextAlignment.center
        nameLabel.numberOfLines = 0
        nameLabel.textColor = UIColor.white
        
        self.addSubview(nameLabel)
    }
    
    func setUserModel(user : UserModel) {
        nameLabel.text = user.getNameString()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
